tests/test_ISComm.cpp

```diff
@@ -1319,3 +1318,255 @@ TEST(ISComm, TruncatedPackets)
 	EXPECT_EQ(g_comm.rxErrorCount, badPktCount);
 }
 #endif
+
+
+#if TEST_BUFF_PARSE_MSG
+
+#define BUFF_PARSE_PASSES		1000000
+#define BUFF_PARSE_OUT_BUF_SIZE 600  
+#define BUFF_PARSE_DEV          0  
+#define BUFF_PARSE_DEV_NMEA     1  
+#define BUFF_PARSE_GPS          2  
+#define BUFF_PARSE_IMU          3  
+#define BUFF_PARSE_INS          4  
+
+static uint32_t s_buffParseMsgInCnt[5] = { 0 };
+
+/**
+ * @brief ISB callback for testing the function is_comm_buffer_parse_messages()
+ */
+int BufferParse_isb(unsigned int port, p_data_t* data)
+{
+	switch (data->hdr.id)
+	{
+        case DID_DEV_INFO:  s_buffParseMsgInCnt[BUFF_PARSE_DEV]++;  break;
+        case DID_INS_2:     s_buffParseMsgInCnt[BUFF_PARSE_INS]++;  break;
+        case DID_GPS1_POS:  s_buffParseMsgInCnt[BUFF_PARSE_GPS]++;  break;
+        case DID_IMU:       s_buffParseMsgInCnt[BUFF_PARSE_IMU]++;  break;
+	}
+
+	return 0;
+}
+
+/**
+ * @brief NMEA callback for testing the function is_comm_buffer_parse_messages()
+ */
+int BufferParse_nmea(unsigned int port, const unsigned char* msg, int msgSize)
+{
+	switch (getNmeaMsgId(msg, msgSize))
+    {
+        case NMEA_MSG_ID_INFO:  s_buffParseMsgInCnt[BUFF_PARSE_DEV_NMEA]++;  break;
+    }
+    return 0;
+}
+
+/**
+ * @brief Tests and exercises the function is_comm_buffer_parse_messages()
+ */
+TEST(ISComm, BufferParse)
+{
+    is_comm_instance_t comm;
+
+    is_comm_callbacks_t callbacks = {};
+	callbacks.isbData = BufferParse_isb;
+	callbacks.nmea = BufferParse_nmea;
+
+    uint32_t msgOutCnt[5] = {0};
+    uint8_t randomBuf[16] = {0xe0,0x63,0xa4,0x17,0x95,0xb1,0x26,0xd2,0xe0,0x63,0xa4,0x17,0x95,0xb1,0x26,0xd2};
```

Comment:
Is this random data?  If so, it should probably be randomly generated (dynamically).  Also, a lot of the following code appears to be "test data".  I think we have some routines that are responsible for actually generating test data, with some random parameters, etc.

At best, you should be trying to use those functions. Otherwise, we should try and make this reuseable, so that if we need custom data like this elsewhere, it can be reused.